### PR TITLE
update dependency to volcano.sh/api

### PR DESCRIPTION
**Pull request by Thor-wl**
Signed-off-by: Thor-wl <13164644535@163.com>

### CODE REVIEW

File: vendor/volcano.sh/apis/pkg/apis/batch/v1alpha1/job.go [hunk 1]
```diff
@@ -29,16 +29,6 @@ import (
...
-// +kubebuilder:printcolumn:name="Pending",type=integer,JSONPath=`.status.pending`
-// +kubebuilder:printcolumn:name="Running",type=integer,JSONPath=`.status.running`
-// +kubebuilder:printcolumn:name="Succeeded",type=integer,JSONPath=`.status.succeeded`
-// +kubebuilder:printcolumn:name="Failed",type=integer,JSONPath=`.status.failed`
-// +kubebuilder:printcolumn:name="Terminating",type=integer,JSONPath=`.status.terminating`
-// +kubebuilder:printcolumn:name="Unknown",type=integer,JSONPath=`.status.unknown`
-// +kubebuilder:printcolumn:name="RetryCount",type=integer,JSONPath=`.status.retryCount`
-// +kubebuilder:printcolumn:name="MinAvailable",type=integer,JSONPath=`.status.minAvailable`
-// +kubebuilder:printcolumn:name="State",type=string,JSONPath=`.status.state.phase`
-// +kubebuilder:printcolumn:name="Age",type=date,JSONPath=`.metadata.creationTimestamp`
```
Comment (shinytang6) [inline vendor/volcano.sh/apis/pkg/apis/batch/v1alpha1/job.go:41]: These annotations are newly added features, we can update to volcano/apis :)

File: vendor/modules.txt [hunk 2]
```diff
@@ -715,3 +743,4 @@ sigs.k8s.io/yaml
 # k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.18.3
 # k8s.io/sample-cli-plugin => k8s.io/sample-cli-plugin v0.18.3
 # k8s.io/sample-controller => k8s.io/sample-controller v0.18.3
+# volcano.sh/apis => volcano.sh/apis v1.3.0-k8s1.18.3-alpha.3
```
Comment (shinytang6) [inline vendor/modules.txt:746]: could we remove the vendor directory, l think it is not necessary if we use go mod?

Comment (Thor-wl) [reply vendor/modules.txt:746]: > could we remove the vendor directory, l think it is not necessary if we use go mod?

Of course.

**Comment by zen-xu** [comment]
What's the benefit of splitting `api` into another repo.

**Comment by Thor-wl** [comment]
> What's the benefit of splitting `api` into another repo.

Downstream community such as kubeflow can get CRD clientset generated by certain K8s version. It is separated with the main repository.

File: vendor/modules.txt (hunk 2 above)
Comment (william-wang) [reply vendor/modules.txt:746]: we'd better keep vendor. It is convient and no need download dependent package everytime.

File: config/crd/bases/batch.volcano.sh_jobs.yaml [hunk 3]
```diff
@@ -19,38 +19,7 @@ spec:
     singular: job
   scope: Namespaced
   versions:
-  - additionalPrinterColumns:
```
Comment (william-wang) [inline config/crd/bases/batch.volcano.sh_jobs.yaml:22]: These changes is commited to volcano.sh/volcano/pkg/api/ after you move api to volcano/api/.  Need to update the chagnes to volcano.sh/api master first and then refresh this pr.

**Review by william-wang — CHANGES_REQUESTED** [review]

**Pull request merged**